crates/walrus-service/src/client/cli/internal_run.rs

```diff
@@ -0,0 +1,766 @@
+
+                    return;
+                }
+            },
+            Err(e) => {
+                tracing::debug!(%e, line = line, "child: failed to parse JSON line");
+            }
+        }
+    }
+
+    for (_, pb) in per_blob_bars.into_iter() {
+        pb.finish_and_clear();
+    }
+}
+
+/// Applies the common store child arguments to the command.
+/// This is used to configure the child process for the store command.
+fn apply_common_store_child_args(
+    cmd: &mut TokioCommand,
+    epoch_arg: &EpochArg,
+    dry_run: bool,
+    store_optimizations: &StoreOptimizations,
+    persistence: BlobPersistence,
+    post_store: PostStoreAction,
+    upload_mode: Option<UploadMode>,
+) {
+    if let Some(ref epochs) = epoch_arg.epochs {
+        match epochs {
+            super::args::EpochCountOrMax::Max => {
+                cmd.arg("--epochs").arg("max");
+            }
+            super::args::EpochCountOrMax::Epochs(count) => {
+                cmd.arg("--epochs").arg(count.to_string());
+            }
+        }
+    }
+
+    if let Some(time) = &epoch_arg.earliest_expiry_time {
+        let datetime: DateTime<Utc> = (*time).into();
+        cmd.arg("--earliest-expiry-time").arg(datetime.to_rfc3339());
+    }
+
+    if let Some(end_epoch) = &epoch_arg.end_epoch {
+        cmd.arg("--end-epoch").arg(end_epoch.to_string());
+    }
+
+    if dry_run {
+        cmd.arg("--dry-run");
+    }
+
+    if !store_optimizations.check_status {
+        cmd.arg("--force");
+    }
+
+    if !store_optimizations.reuse_resources {
+        cmd.arg("--ignore-resources");
+    }
+
+    match persistence {
+        BlobPersistence::Deletable => cmd.arg("--deletable"),
+        BlobPersistence::Permanent => cmd.arg("--permanent"),
+    };
+
+    if post_store == PostStoreAction::Share {
+        cmd.arg("--share");
+    }
+
+    if let Some(mode) = upload_mode {
+        cmd.arg("--upload-mode").arg(match mode {
+            UploadMode::Conservative => "conservative",
+            UploadMode::Balanced => "balanced",
+            UploadMode::Aggressive => "aggressive",
+        });
+    }
+}
+
+/// Spawns a child process to handle the upload of blobs.
+/// This is used to handle the upload of blobs in a separate child process.
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn maybe_spawn_child_upload_process<F>(
+    client: &WalrusNodeClient<SuiContractClient>,
+    epoch_arg: &EpochArg,
+    dry_run: bool,
+    store_optimizations: &StoreOptimizations,
+    persistence: BlobPersistence,
+    post_store: PostStoreAction,
+    upload_mode: Option<UploadMode>,
+    upload_relay: Option<&Url>,
+    internal_run: bool,
+    command_name: &str,
+    add_command_args: F,
+    num_blobs: usize,
+) -> Result<Option<()>>
+where
+    F: FnOnce(&mut TokioCommand),
+{
+    let child_uploads_enabled = client.config().communication_config.child_uploads_enabled;
+    if !(child_uploads_enabled && upload_relay.is_none() && !internal_run) {
```

Comment:
Question: I'm not sure if I fully understand the condition. It seems that !child_upload_enabled || internal_run would not use child process?

Also, can we include client.config().communication_config.child_uploads_enabled when setting internal_run? Seems they are talking about the same thing: using child process to upload.